### PR TITLE
Merge dev to main for release 1.3.0

### PR DESCRIPTION
**Pull request by berndfinger**

### CODE REVIEW

File: .github/workflows/ansible-lint-sap_general_preconfigure.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_general_preconfigure.yml:27]:
```suggestion
        uses: actions/checkout@v3
```
should be v3 instead of main because of main is used for upstream

File: .github/workflows/ansible-lint-sap_general_preconfigure.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_general_preconfigure.yml:30]:
```suggestion
        uses: actions/setup-python@v4
```
same argument as above

File: .github/workflows/ansible-lint-sap_ha_install_hana_hsr.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_ha_install_hana_hsr.yml:27]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/ansible-lint-sap_ha_install_hana_hsr.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_ha_install_hana_hsr.yml:30]:
```suggestion
        uses: actions/setup-python@v4
```

File: .github/workflows/ansible-lint-sap_ha_pacemaker_cluster.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_ha_pacemaker_cluster.yml:27]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/ansible-lint-sap_ha_pacemaker_cluster.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_ha_pacemaker_cluster.yml:30]:
```suggestion
        uses: actions/setup-python@v4
```

File: .github/workflows/ansible-lint-sap_hana_install.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_hana_install.yml:27]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/ansible-lint-sap_hana_install.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_hana_install.yml:30]:
```suggestion
        uses: actions/setup-python@v4
```

File: .github/workflows/ansible-lint-sap_hana_preconfigure.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_hana_preconfigure.yml:27]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/ansible-lint-sap_hana_preconfigure.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_hana_preconfigure.yml:30]:
```suggestion
        uses: actions/setup-python@v4
```

File: .github/workflows/ansible-lint-sap_hypervisor_node_preconfigure.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_hypervisor_node_preconfigure.yml:27]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/ansible-lint-sap_hypervisor_node_preconfigure.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_hypervisor_node_preconfigure.yml:30]:
```suggestion
        uses: actions/setup-python@v4
```

File: .github/workflows/ansible-lint-sap_netweaver_preconfigure.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_netweaver_preconfigure.yml:27]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/ansible-lint-sap_netweaver_preconfigure.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_netweaver_preconfigure.yml:30]:
```suggestion
        uses: actions/setup-python@v4
```

File: .github/workflows/ansible-lint-sap_storage_setup.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_storage_setup.yml:27]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/ansible-lint-sap_storage_setup.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_storage_setup.yml:30]:
```suggestion
        uses: actions/setup-python@v4
```

File: .github/workflows/ansible-lint-sap_swpm.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_swpm.yml:27]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/ansible-lint-sap_swpm.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_swpm.yml:30]:
```suggestion
        uses: actions/setup-python@v4
```

File: .github/workflows/ansible-lint-sap_vm_preconfigure.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_vm_preconfigure.yml:27]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/ansible-lint-sap_vm_preconfigure.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint-sap_vm_preconfigure.yml:30]:
```suggestion
        uses: actions/setup-python@v4
```

File: .github/workflows/ansible-lint.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint.yml:19]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/ansible-lint.yml
Comment (rainerleber) [inline .github/workflows/ansible-lint.yml:22]:
```suggestion
        uses: actions/setup-python@v4
```

File: .github/workflows/ansible-test-sanity.yml
Comment (rainerleber) [inline .github/workflows/ansible-test-sanity.yml:19]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/ansible-test-sanity.yml
Comment (rainerleber) [inline .github/workflows/ansible-test-sanity.yml:22]:
```suggestion
        uses: actions/setup-python@v4
```

File: CHANGELOG.rst
Comment (rainerleber) [inline CHANGELOG.rst:15]:
```suggestion
| Release Date: 2023-07-21
```

**Review by rainerleber — COMMENTED** [review]
just a few suggestions for the workflows

**Comment by rainerleber** [comment]
I think the @main reference for the workflows is to simplify maintaining ? But because the main branch is used as upstream it could lead into issues. Maybe a better idea is to use renovatebot to auto maintain changes in the actions.

**Comment by rainerleber** [comment]
Maybe we should discuss the branching and staging strategy as well. Because such big PRs are hard to review.

Suggestion:
- Remove dev Branch
- Create PR to main for each change
   - like one pr for:
      storage role changes
      workflow changes
      doc changes
      pacemaker role changes
- Create Changelog Fragments for each pr

Advantage PRs are reviewed per dedicated change
Less to review at once.
Faster Release because changes are reviewed before merge.
....

It's a good practice from the Ansible BU too.

**Comment by ja9fuchs** [comment]
> Maybe we should discuss the branching and staging strategy as well. Because such big PRs are hard to review.
> 
> Suggestion:
> 
> * Remove dev Branch
> * Create PR to main for each change
>   
>   * like one pr for:
>     storage role changes
>     workflow changes
>     doc changes
>     pacemaker role changes

We are mostly doing this already, but against the `dev` branch.

> * Create Changelog Fragments for each pr
> 
> Advantage PRs are reviewed per dedicated change Less to review at once. Faster Release because changes are reviewed before merge. ....

We have already been reviewing all those changes in separate PRs that were raised against `dev` first.
After the dev merge, it is easier to use `dev` as a staging area and run larger tests including changes from different roles. 
We identify and address issues in `dev` on a regular basis that have slipped through PR review, simply because of the complexity of this collection. I thought this was what a `dev` branch is used for - to test before changes go into the `main` branch, which would be considered "stable"?

Maybe I'm too strict with my understanding of a stable `main` branch between releases.

Obviously, the less frequent merge of `dev` into `main` is a larger step. However, at this stage it is much easier to run a test including multiple roles against `dev`, compared to testing many PRs with the changes still sitting in different and unrelated personal forks.

I agree, one huge PR is not great, even though we have reviewed and tested the changes already in the staging phase. 

> It's a good practice from the Ansible BU too.

We should put the topic on the meeting agenda and discuss possible alternative workflows and potential effects.

**Comment by ja9fuchs** [comment]
@rainerleber 
I made these changes when investigating the workflows failing for over 2 weeks in a row and they have been working well since. I believe this is not a merge blocking priority at this moment.
Would you agree to resolve the suggestions and create an issue instead, so that we can discuss the workflow maintenance and GH action versions separately later and without blocking this PR?

**Comment by sean-freeman** [comment]
@rainerleber as a brief addition to above comments.... this release contains big functionality improvements for 3 Ansble Roles simultaneously (Storage, Media, HA); it is the reason for the minor version number increment from `1.2.z` > `1.3.z`.

We were actively concerned about another large PR (like before with `1.1.z` > `1.2.z` with ~500 commits) and attempted to avoid it. Even patch releases have a fair amount of commits (like before 1.2.2 > 1.2.3 with ~50 commits). The practicality of `dev` branch being our single staging/integration catching bugs and functionality improvements means larger PRs for minor version increments. The balancing act is difficult, I would always prefer that we keep `dev` branch as-is because as said - we do catch accidental bugs or changes that create bugs elsewhere further down the line. All core developers use and test against `dev` branch.

I would suggest the GH Action Workflow version lock be moved into a separate GH Issue for the next patch release v1.3.1 and not be addressed now. There is improvements needed to the GH Action Workflow as a whole, in my opinion.

**Comment by rainerleber** [comment]
I think it's the wrong way to discuss and will open a discussion for this :-). Beyond my suggestions I don't have any concerns so lgtm

**Review by rainerleber — APPROVED** [review]

File: .github/workflows/ansible-lint-sap_general_preconfigure.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_general_preconfigure.yml:27]: Closing for now - we will continue the discussion in https://github.com/sap-linuxlab/community.sap_install/discussions.

File: .github/workflows/ansible-lint-sap_ha_install_hana_hsr.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_ha_install_hana_hsr.yml:27]: Closign for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_ha_install_hana_hsr.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_ha_install_hana_hsr.yml:30]: Closign for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_ha_pacemaker_cluster.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_ha_pacemaker_cluster.yml:27]: Closign for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint.yml:22]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_ha_pacemaker_cluster.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_ha_pacemaker_cluster.yml:30]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_hana_install.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_hana_install.yml:27]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_hana_install.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_hana_install.yml:30]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_hana_preconfigure.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_hana_preconfigure.yml:30]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_hypervisor_node_preconfigure.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_hypervisor_node_preconfigure.yml:27]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_hypervisor_node_preconfigure.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_hypervisor_node_preconfigure.yml:30]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_netweaver_preconfigure.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_netweaver_preconfigure.yml:27]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_netweaver_preconfigure.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_netweaver_preconfigure.yml:30]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_storage_setup.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_storage_setup.yml:27]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_storage_setup.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_storage_setup.yml:30]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_swpm.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_swpm.yml:27]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_vm_preconfigure.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_vm_preconfigure.yml:27]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_vm_preconfigure.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_vm_preconfigure.yml:30]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-test-sanity.yml
Comment (berndfinger) [reply .github/workflows/ansible-test-sanity.yml:19]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-test-sanity.yml
Comment (berndfinger) [reply .github/workflows/ansible-test-sanity.yml:22]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

File: .github/workflows/ansible-lint-sap_swpm.yml
Comment (berndfinger) [reply .github/workflows/ansible-lint-sap_swpm.yml:30]: Closing for now, see https://github.com/sap-linuxlab/community.sap_install/pull/419#discussion_r1270607762.

**Review by sean-freeman — APPROVED** [review]
Extensively tested, approved for merge 👍

**Pull request merged**
